File: GO/1.go
func twoSum(nums []int, target int) []int {
	//双重循环
	// for i := 0; i < len(nums); i++ {
	//     for j := i+1; j < len(nums); j++ {
	//         if nums[i] + nums[j] == target {
	//             return []int{i, j}
	//         }
	//     }
	// }
	// return []int{0, 0}

	//hash
	hashTable := map[int]int{}
	for index1, value1 := range nums {
		index2, ok := hashTable[value1]
		if ok {
			return []int{index1, index2}
		}
		hashTable[target-value1] = index1
	}
	return []int{}
}

File: 1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int len = nums.length;
        for(int i = 0;i < len;i ++) {
            if(map.containsKey(target - nums[i])) return new int[]{map.get(target - nums[i]), i};
            map.put(nums[i], i);
        }
        return new int[]{0, 0};
    }
}